### DISCUSSION TITLE
Drives show as Incompatible after running script on  DS3622sx+

### DISCUSSION THREAD

**Discussion by Perkinje** [007revad/Synology_HDD_db#363] [Q&A]
I placed the script, hdd vendor file, and bin folder in /volume1/scripts and set the script as executable. I then created a task in the control panel to run as root at boot-up using 
```
sudo -s /volume1/scripts/syno_hdd_db.sh -nr --autoupdate=90 2>&1 | tee /volume1/scripts/syno-hd-db.log
``` 
I initially tried it without the r and sudo -s options and without pushing the output to a log file. Rebooted and now the drives show as incompatible.

![Screenshot 2024-10-11 at 10 53 57 AM](https://github.com/user-attachments/assets/c671cde0-de6f-41ca-9f41-a972026dab64)

Output from the script is 
```Synology_HDD_db v3.5.102
DS3622xs+ DSM 7.2.2-72806 
StorageManager 1.0.0-00502

ds3622xs+_host_v7 version 8041

Using options: -nr --autoupdate=90
Running from: /volume1/scripts/syno_hdd_db.sh

HDD/SSD models found: 4
HUH721010ALE604,LHAZW2J2,10000 GB
ST16000NM001G-2KK103,SN03,16000 GB
ST18000NM003D-3DL103,SN03,17999 GB
ST20000NM007D-3DJ103,SN01,19999 GB

M.2 drive models found: 1
Sabrent Rocket Q,RKT30Q.3,1000 GB

M.2 PCIe card models found: 1
M2D20

No Expansion Units found

HUH721010ALE604 already exists in ds3622xs+_host_v7.db
ST16000NM001G-2KK103 already exists in ds3622xs+_host_v7.db
ST18000NM003D-3DL103 already exists in ds3622xs+_host_v7.db
ST20000NM007D-3DJ103 already exists in ds3622xs+_host_v7.db
Sabrent Rocket Q already exists in ds3622xs+_host_v7.db
Sabrent Rocket Q already exists in ds3622xs+_m2d20_v7.db

M2D20 NVMe already enabled for DS3622xs+

Support disk compatibility already enabled.

Disabled support memory compatibility.

Set max memory to 64 GB.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already disabled.

Creating pool in UI on drives in M.2 adaptor card already enabled.

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.
```

I've tried the -f and -i options with the same outcome. I've also rebooted between each attempt. Any help would be appreciated.

**Comment by 007revad** [comment]
Can you download a copy of the following files: 
1. /etc/model.dtb
2. /etc.defaults/model.dtb
3. /run/model.dtb

Then zip them and attach the zip file to your reply.

**Reply by Perkinje** [reply]
I do not have any of those files.
```
~$ ls /etc/m*
/etc/machine-id  /etc/mke2fs.conf  /etc/modules.conf  /etc/motd  /etc/mtab  /etc/mt-daapd.playlist

/etc/mcelog:
bus-error-trigger    dimm-error-trigger   mcelog.conf                             page-error-post-sync-soft-trigger  page-error-trigger           unknown-error-trigger
cache-error-trigger  iomca-error-trigger  page-error-counter-replacement-trigger  page-error-pre-sync-soft-trigger   socket-memory-error-trigger
```
```
~$ ls /etc.defaults/m*
/etc.defaults/mke2fs.conf  /etc.defaults/modules.conf  /etc.defaults/motd  /etc.defaults/mtab

/etc.defaults/mcelog:
bus-error-trigger    dimm-error-trigger   mcelog.conf                             page-error-post-sync-soft-trigger  page-error-trigger           unknown-error-trigger
cache-error-trigger  iomca-error-trigger  page-error-counter-replacement-trigger  page-error-pre-sync-soft-trigger   socket-memory-error-trigger
```
```
~$ ls /run/m*
/run/mcelog-client  /run/mediaserver_sh.lock     /run/mib_log_uploader.pid      /run/mib_monitor.pid                     /run/mib_status_uploader.pid            /run/minissdpd.pid
/run/mcelog.pid     /run/mib_event_uploader.pid  /run/mib_metrics_uploader.pid  /run/mib_server_status_cache_daemon.pid  /run/mib_upgrade_progress_uploader.pid  /run/minissdpd.sock

/run/mdadm:
map

/run/mib_disklatency:
syno-disk-latency-anchor-sda  syno-disk-latency-anchor-sdc  syno-disk-latency-anchor-sde  syno-disk-latency-anchor-sdg
syno-disk-latency-anchor-sdb  syno-disk-latency-anchor-sdd  syno-disk-latency-anchor-sdf  syno-disk-latency-anchor-sdi
```

**Reply by 007revad** [reply]
Oops, sorry.

That should have been: 
- /var/lib/disk-compatibility/ds3622xs+_host_v7.db
- /var/lib/disk-compatibility/ds3622xs+_m2d20_v7.db

**Comment by Perkinje** [comment]
[ds3622sx+_host_m2d20_v7.db.zip](https://github.com/user-attachments/files/17348221/ds3622sx%2B_host_m2d20_v7.db.zip)

**Comment by Perkinje** [comment]
I just altered the rules and eunit files. Drive status is now "Healthy", but I still see this...
![Screenshot 2024-10-11 at 5 21 42 PM](https://github.com/user-attachments/assets/082c65db-36b7-491d-9ae1-a6fe102e984e)
and 
![Screenshot 2024-10-11 at 5 21 09 PM](https://github.com/user-attachments/assets/477a1252-a217-4259-99d5-ed1f0f8cec2c)

**Reply by 007revad** [reply]
I assume you mean the host_rule.db and eunit_rule.db files in /var/lib/disk-compatibility/rules ?

What did you change?

**Reply by Perkinje** [reply]
I swapped them for these.
[host_eunit_rules.zip](https://github.com/user-attachments/files/17348647/host_eunit_rules.zip)
I made a backup of the originals, though.

**Reply by Perkinje** [reply]
Were you able to review the files and see where I messed this up?

**Reply by 007revad** [reply]
Your files looked okay when I first checked them. But I just had another look and noticed the size of 2 of your HDDs are wrong. 17999 instead of 1800 and 19999 instead of 20000.

Can you try the following: 
1. Run syno_hdd_db with the --restore option.
2. Update to https://github.com/007revad/Synology_HDD_db/releases/tag/v3.5.103 and run it.

**Reply by Perkinje** [reply]
They all show as "Incompatible" again, but the HDD sizes are now correct in the log output.

**Reply by 007revad** [reply]
Is this a real Synology or xpenology?

Assuming you have the script scheduled to run at bootup try rebooting once or twice. Some people have reported they needed to run the script and reboot 2 or 3 times.

**Reply by Perkinje** [reply]
Genuine Synology device and 4 reboots without change.

**Reply by 007revad** [reply]
Sorry, I got side-tracked and forgot about you. Are you still having an issue?
